File: Project-Sort-Testing/快速排序/C++/QuickSort/QuickSort.cpp
// QuickSort.cpp : 定义控制台应用程序的入口点。
//

#include "stdafx.h"
#include <iostream>
using namespace std;

int a[10]={5,7,1,6,8,2,9,3,0,4};
int Partition(int l,int r)
{
	int i,j,x,tmp;
	i=l-1; x=a[r];
	for (j=l;j<r;j++)
	{
		while (i<j && a[j]<=x)
		{
			i++;
			tmp=a[i];
			a[i]=a[j];
			a[j]=tmp;
		}
	}
	tmp=a[i+1];
	a[i+1]=a[r];
	a[r]=tmp;
	return i+1;
}

void QuickSort(int l,int r)
{
	if (l<r) 
	{
		int x=Partition(l,r);
		QuickSort(l,x-1);
		QuickSort(x+1,r);
	}
}
int _tmain(int argc, _TCHAR* argv[])
{
	int i;
	cout<<"Before Sort"<<endl;
	for(i=0;i<10;i++) cout<<a[i];
	cout<<endl;
	QuickSort(0,9);
	cout<<"After Sort"<<endl;
	for(i=0;i<10;i++) cout<<a[i];
	cout<<endl;
	cin>>i;
	return 0;
}

File: Project-Sort-Testing/测评/OFFICE 03/算法测评/快速排序/pascal/quicksort.pas
program quicksort;
uses sysutils;
const ArrMax=100;
var a:array[1..ArrMax] of integer;
k,x,s:integer;
time:real;

function Partition(l,r:integer):integer;
var i,j,tmp,x:integer;
begin
	i:=l-1;x:=a[r];
	for j:=l to r-1 do
	begin
		while (i<j) and (a[j]<=x) do
		begin
			i:=i+1;
			tmp:=a[i];
			a[i]:=a[j];
			a[j]:=tmp;
		end;
	end;
	tmp:=a[i+1];
	a[i+1]:=a[r];
	a[r]:=tmp;
	Partition:=i+1;
end;

procedure QucikSort(l,r:integer);
var tmp:integer;
begin
	if (l<r) then
	begin
		tmp:=Partition(l,r);
		QucikSort(l,tmp-1);
		QucikSort(tmp+1,r);
	end;
end;

begin
time:=now;
for x:=1 to 100 do  //TODO:Set the times to loop
begin
	randomize;
	for s:=1 to ArrMax do a[s]:=random(1000);
	writeln('Before Sort');
	for k:=1 to ArrMax do write(a[k]:5);
	writeln;
	QucikSort(1,ArrMax);
	writeln('After Sort');
	for k:=1 to ArrMax do write(a[k]:5);
	writeln;
end;
writeln((now-time)*86400);
readln;
end.